### PR TITLE
Clean up chat functions

### PR DESCRIPTION
**Pull request by aokeke**
Add some test cases for the chat functionality and make the tagRemovalRefs no longer global.

### CODE REVIEW

File: proj/src/main/webapp/test/chat.test.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+            await firebase.auth().signOut()
+        }
+    });
+
+    after(async function() {
+        if (testconfig.shouldCleanUp) {
+            // // Delete the test users
+            // for (let i=0;i<testUsers.length;i++){
+            //     await deleteUser(testUsers[i])
+            // }
```
Comment (aokeke) [inline proj/src/main/webapp/test/chat.test.js]: Considering if we should just never delete some test users so they can be reused. If you delete and recreate users too many times, firebase starts blocking your IP.

**Review by nlumiere — CHANGES_REQUESTED** [review]
Pretty sure there are gonna be some conflicts with current master

File: proj/src/main/webapp/static/chat.html [hunk 2]
```diff
@@ -157,7 +158,7 @@ <h1 class="display-4 card-header" id="big-title"> Chat Room </h1>
                                     </div>
                                     <div calss="col-11">
                                         <li class="list-group-item">
-                                            <div class=" justify-content-between">
+                                            <div class="message-header justify-content-between">
```
Comment (leeannachen) [inline proj/src/main/webapp/static/chat.html]: added this line to my version and it fixed all the bugs 👍

**Review by leeannachen — COMMENTED** [review]
lgtm

File: proj/src/main/webapp/chat.js [hunk 3]
```diff
@@ -236,24 +231,27 @@ async function setUserTags(tagList){
...
                 }
                 else{
-                    key = currentTags[tag];
+                    allTags[tag] = currentTags[tag];
+                    tagRemovalDict[tag] = currentTagRemovalDict[tag];
                     delete currentTags[tag];
+                    delete currentTagRemovalDict[tag];
                 }
-                allTags[tag] = key;
             }
```
Comment (nlumiere) [inline proj/src/main/webapp/chat.js:181]: This looks like it should work, but it's pretty convoluted still. I'm good with it as long as it's tested and confirmed to work.

**Review by nlumiere — APPROVED** [review]
I don't see why anything shouldn't work... Though lease test the more complicated things if you haven't already.

**Review by arringtonh — APPROVED** [review]
lgtm!

**Pull request merged**